### DISCUSSION TITLE
Doubt regarding Input file structure for MAW

### DISCUSSION THREAD

**Discussion by Anurag19101996** [uw-ssec/MAWpy#37] [General]
In the [One note ](https://uwnetid.sharepoint.com/sites/og_ssec_escience/_layouts/15/Doc.aspx?sourcedoc=%7B6b0ea251-f0a8-4ce3-8ea9-d1d796dcf28f%7D&action=edit&wd=target%28Meeting%20Notes.one%7Cb0d0aa65-2fb3-4a83-9b78-ca0f174e22b5%2F2024.06.20%20UW%20internal%20meeting%7C4f2f25d4-7499-4f2c-aa8a-252b79a0cfdf%2F%29&wdorigin=NavigationUrl)file, we want the MAW code to take inputs that have four mandatory columns (user ID, date time, orig_lat, orig_long) and other non mandatory columns like orig_unc. The incremental clustering code (src/mawpy/steps/incremental_clustering.py) still assumes the input file to have columns that do not belong to the four mandatory columns. For example, line 26 says “from mawpy.constants import USER_ID, STAY_DUR, ORIG_LAT, STAY_LAT, STAY_LONG, STAY_UNC, ORIG_LONG, ORIG_UNC, STAY”, which assumes extra columns,  such as stay_lat, stay_long, stay_unc, etc. are part of the input data. 

a. This could work out if these extra columns are manually added to the input file by the MAW algorithm, but we do not see that code of manual insertion. Is that something planned by Escience later down the road? If not, the logic here does not seems like Thinklab’s request is being met.

b. Also, as an example For function _get_combined_stay, we have values grabbed from columns STAY, STAY_LAT, and STAY_LONG. The problem here is that the input files might not have those columns to begin with thus no values to grab.

@qzchen-uw @gracejia513 @anujsinha3 @carlosgjs

**Comment by anujsinha3** [comment]
The code only expects `user_id,` `orig_lat`, `orig_long`, `orig_unc`(non-mandatory) and `unix_start_t`. 

(src/mawpy/steps/incremental_clustering.py) DOES NOT expect any other columns as part of input csv.

`STAY_LAT, STAY_LONG, STAY_UNC, STAY` are columns for which values are calculated by incremental_clustering code itself. These columns are not expected to be part of input csv.

In python, `import` statements are just a way to use variables/methods that are part of another python file. Hence, the statement below does not directly relate to expected input columns:

`import USER_ID, STAY_DUR, ORIG_LAT, STAY_LAT, STAY_LONG, STAY_UNC, ORIG_LONG, ORIG_UNC, STAY`.

Hope this clarifies.
